test/functional/services/apps_menu.ts

```diff
@@ -116,17 +117,50 @@ export class AppsMenuService extends FtrService {
       category,
     }: { closeCollapsibleNav?: boolean; category?: string } = {}
   ) {
+    await this.waitUntilLoadingHasFinished();
+
     try {
       this.log.debug(`click "${name}" app link`);
       await this.openCollapsibleNav();
       let nav;
       if (typeof category === 'string') {
+        // we can search within a specific section of the side nav
         nav = await this.testSubjects.find(`collapsibleNavGroup-${category}`);
+        const link = await nav.findByPartialLinkText(name);
+        await link.click();
```

Comment:
Potentially it can be a place to catch StaleElementException: 
1. We search for element by name in line 129 and safe element referenceID of this element in DOM
2. If for some reason navigation component is re-rendered, the previous referenceID is no longer valid and click will fail.

If re-rendering is unlikely to happen, please ignore my comment.
If it sounds like a possible behaviour while using `appsMenu.clickLink()` there are 2 ways:
1) wrap it up with retry
2) use css/xpath selector to find element directly, not inside `nav` object

Reply:
Thanks! I just wrapped it with `retry` to account for stale elements.